private_sharing/forms.py

```diff
@@ -397,3 +419,32 @@ class DeleteDataFileForm(forms.Form):
     file_basename = forms.CharField(required=False, label="File basename")
 
     all_files = forms.BooleanField(required=False, label="All files")
+
+
+class SelectDatatypesForm(forms.Form):
+    """
+    A form that generates a tree for selecting datatypes.
+    """
+
+    def clean(self):
+        cleaned_data = super().clean()
+        # Check to see if anything was selected
+        # wants to be on two lines:
+        data = dict(self.data)
+        data.pop("csrfmiddlewaretoken")
+        if data:
+            cleaned_data.update(self.data)
+            self.cleaned_data = data
```

Comment:
is this overriding/redoing stuff that super() already does?

Maybe this method could be reduced to simply raise a `ValidationError` if at least one hasn't been selected? Otherwise, return the super()?

Reply:
cleaned_data.update(self.data) is really not necessary here; otherwise returning the dict without the csrf token does make processing later easier.

the super() does not actually populate cleaned_data if there are no fields formally defined, making this override necessary.